src/main.c

```diff
@@ -394,36 +429,34 @@ int main(int argc, char* argv[]) {
           fprintf(stderr, "%s: --%s takes two parameters (e.g. --%s varname filename)\n", progname, which, which);
           die();
         }
-        jv arg = jv_object();
-        arg = jv_object_set(arg, jv_string("name"), jv_string(argv[i+1]));
-        jv data = jv_load_file(argv[i+2], 0);
-        if (!jv_is_valid(data)) {
-          data = jv_invalid_get_msg(data);
-          fprintf(stderr, "%s: Bad JSON in --%s %s %s: %s\n", progname, which,
-                  argv[i+1], argv[i+2], jv_string_value(data));
-          jv_free(data);
-          jv_free(arg);
-          ret = 2;
-          goto out;
+        if (!jv_object_has(jv_copy(program_arguments), jv_string(argv[i+1]))) {
+          jv data = jv_load_file(argv[i+2], 0);
+          if (!jv_is_valid(data)) {
+            data = jv_invalid_get_msg(data);
+            fprintf(stderr, "%s: Bad JSON in --%s %s %s: %s\n", progname, which,
+                    argv[i+1], argv[i+2], jv_string_value(data));
+            jv_free(data);
+            ret = 2;
+            goto out;
+          }
+          if (isoption(argv[i], 0, "argfile", &short_opts) &&
```

Comment:
Fair enough, but I saw you add the `which`, so I thought we could improve while we were here. 😄